indico/modules/users/util.py

```diff
@@ -422,7 +422,26 @@ def merge_users(source, target, force=False):
     # Restore the source user's primary email
     source.email = primary_source_email
     db.session.flush()
+    
+    log_dict = {
+        'module': 'User',
+        'message': 'User merged',
+        'user': session.user if session else None
+    }
+    
+    def get_user_log_data(user, id_key):
+        return {
+            id_key: user.id,
+            'First Name': user.first_name,
+            'Last Name': user.last_name
+        }
+
+    data_target = get_user_log_data(source, 'Source ID')
+    data_source = get_user_log_data(target, 'Target ID')
 
+    target.log(UserLogRealm.user, LogKind.other, **log_dict, data=data_target)
```

Comment:
You can include `realm` and `kind` in the dict as well. I'd also rename it to `log_args` which is clearer IMHO.